### ISSUE TITLE
Lost compatibility with `actix_web_httpauth`

### ISSUE THREAD

**Issue by ManevilleF** [paperclip-rs/paperclip#404]
Using `actix_web` 4 and paperclip as a git dependency with the new `actix4` feature:

```toml
[dependencies.paperclip]
version = "0.6"
git = "https://github.com/paperclip-rs/paperclip"
branch = "master"
features = ["actix4", "v3", "swagger-ui", "actix-files"]
```

The `paperclip_actix::web::Resource::wrap` method no longer supports the  `actix_web_httpauth::middleware::HttpAuthentication::bearer` middleware.

I get the following error when using it like:

```rust
.service(web::resource("/route/{id}")
         .wrap(HttpAuthentication::bearer(bearer_token_middleware))
         .route(web::get().to(my_controller::show))
)
```

```
error[E0271]: type mismatch resolving `<HttpAuthentication<BearerAuth, fn(ServiceRequest, BearerAuth) -> impl std::future::Future<Output = Result<ServiceRequest, actix_web::Error>> {bearer_token_middleware}> as Transform<actix_web::resource::ResourceService, ServiceRequest>>::Response == ServiceResponse`
   --> file.rs
    |
94  |                     .wrap(HttpAuthentication::bearer(bearer_token_middleware))
    |                      ^^^^ expected struct `BoxBody`, found enum `EitherBody`
    |
    = note: expected struct `ServiceResponse<BoxBody>`
               found struct `ServiceResponse<EitherBody<_>>`
note: required by a bound in `paperclip::actix::web::Resource::<actix_web::Resource<T>>::wrap`
   --> /me/.cargo/git/checkouts/paperclip-30ddbbc2fa493bf0/a0e9d32/plugins/actix-web/src/web.rs:178:17
    |
178 |                 Response = ServiceResponse,
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `paperclip::actix::web::Resource::<actix_web::Resource<T>>::wrap`
    ```

**Comment by tiagolobocastro** [comment]
Tagging @sfisol in case he's able to take a look

**Comment by robjtede** [comment]
Paperclip either needs to wrap everything passed to `.wrap` in `Compat` or support non-`BoxBody` response payload types properly. This is not strictly a blocker to releasing v4 support since the workaround is for users to use `Compat`.

**Comment by tiagolobocastro** [comment]
Thanks @robjtede - yeah `Compat` is fine as a workaround for now.
I'm preparing a new release now.

**Comment by tiagolobocastro** [comment]
Should be fixed by #414
